src/NHibernate.Test/Linq/ConstantTest.cs

```diff
@@ -355,8 +356,90 @@ public void PlansWithNonParameterizedConstantsAreNotCached()
 			 select new { c.CustomerId, c.ContactName, Constant = 1 }).First();
 			Assert.That(
 				cache,
-				Has.Count.EqualTo(0),
-				"Query plan should not be cached.");
+				Has.Count.EqualTo(1),
+				"Query should  be cached.");
+		}
+
+		[Test]
+		public void PlansWithConstantExpressionsAreNotCached()
+		{
+			var queryPlanCacheType = typeof(QueryPlanCache);
+			var cache = (SoftLimitMRUCache) queryPlanCacheType.GetField("planCache", BindingFlags.Instance | BindingFlags.NonPublic).GetValue(Sfi.QueryPlanCache);
+			cache.Clear();
+
+			var input = new { Name = "ALFKI" };
+			(from c in db.Customers
```

Comment:
@gokhanabatay for all these queries you should check for the results returned. Otherwise it might be that not only plan is cached but also results are.

Reply:
> > It seems to me part of the reason is written [here](https://github.com/nhibernate/nhibernate-core/pull/3168#issuecomment-1263573440):
> > > From simple change it's converted to something else. Let's focus on simple fix in this PR and expression types that require additional handling be moved in separate PR.
> 
> I cannot understand, if there is major issues so major changes needs to be applied. We want to fix these problems that we are reporting past 3 years mostly linq performance issues.
> 
> @hazzik @fredericDelaporte Maca just made good improvements my question is what are we waiting, we are volunteered if you guide us we can make necessary changes to merge these pull requests. #2079 and #2375.
> 
> Because our changes are much simpler then maca's, we were repeating its changes without knowing its already done. https://github.com/nhibernate/nhibernate-core/blob/04e40931af0d812810ef176deb452847154b6ee9/src/NHibernate/Linq/Visitors/SelectClauseNominator.cs

@maca88 could you please tell us, what are the necessary changes your pull request neeeds to merge. Or dou you have a plan to work on this. We are talking these issues with you 2019 to today. #2079 and #2375.